include/picongpu/plugins/binning/BinningCreator.hpp

```diff
@@ -52,33 +51,44 @@ namespace picongpu
-             * data dumps and undefined behaviour during restarts
-             * @param axisTupleObject tuple holding the axes
-             * @param speciesTupleObject tuple holding the species to do the binning with
-             * @param depositionData functorDescription of the deposited quantity
-             * @param writeOpenPMDFunctor Functor to write out user specified openPMD data
+             * Creates a particle binner from user input and adds it to the vector of all binners.
+             *
+             * The Particle binner will bin particle quantities to create histograms on a grid defined by axes.
+             * The results will be written to openPMD files.
+             *
+             * @param binnerOutputName filename and dataset name for openPMD output. Must be unique to avoid overwrites
+             * during data dumps and undefined behaviour during restarts.
+             * @param axisTupleObject tuple holding the axes. Each element in the tuple describes an axis for binning.
+             * @param speciesTupleObject tuple holding the species to bin. Each element specifies a species to be
+             * included in the binning.
+             * @param depositionData functor description of the deposited quantity.  Defines which particle property to
+             * bin.
+             * @param extraData tuple holding extra data to be passed to the binner. Can be used for optional
+             * configurations.
+             * @return ParticleBinningData& reference to the created ParticleBinningData object.
+             *         This can be used to further configure the binning setup if needed.
              */
-            template<typename TAxisTuple, typename TSpeciesTuple, typename TDepositionData>
-            auto addBinner(
+            template<
+                typename TAxisTuple,
+                typename TSpeciesTuple,
+                typename TDepositionData,
+                typename T_Extras = std::tuple<>>
+            auto addParticleBinner(
                 std::string const& binnerOutputName,
                 TAxisTuple const& axisTupleObject,
                 TSpeciesTuple const& speciesTupleObject,
                 TDepositionData const& depositionData,
-                std::function<
-                    void(::openPMD::Series& series, ::openPMD::Iteration& iteration, ::openPMD::Mesh& mesh)> const&
-                    writeOpenPMDFunctor
-                = [](::openPMD::Series& series, ::openPMD::Iteration& iteration, ::openPMD::Mesh& mesh) {})
-                -> BinningData<TAxisTuple, TSpeciesTuple, TDepositionData>&
+                T_Extras const& extraData = std::tuple<>{})
+                -> ParticleBinningData<TAxisTuple, TSpeciesTuple, TDepositionData, T_Extras>&
             {
-                auto bd = BinningData<TAxisTuple, TSpeciesTuple, TDepositionData>(
+                auto bd = ParticleBinningData<TAxisTuple, TSpeciesTuple, TDepositionData, T_Extras>(
```

Comment:
```suggestion
                auto bd = ParticleBinningData(
```
Use template argument deduction because, you know... DRY. I've seen somewhere that bad things could happen in this case.